corehq/apps/users/views/utils.py

```diff
@@ -6,14 +6,15 @@
 )
 
 
-def get_editable_role_choices(domain, couch_user, allow_admin_role):
+def get_editable_role_choices(domain, couch_user, allow_admin_role, use_qualified_id=True):
     """
     :param domain: roles for domain
     :param couch_user: user accessing the roles
     :param allow_admin_role: to include admin role, in case user is admin
+    :param use_qualified_id: use role's qualified id as the id for the choice else the db id
```

Comment:
What's the purpose of the qualified id?

Reply:
that was used in the original method that i extracted out..it has a format like `'user-role:%s' % self.get_id`

Reply:
But why does some code use that prefix and other code doesn't?

Reply:
Looks like this is a way to make the non-existent "admin" role existent by adding a prefix for roles and leaving the "admin" one [as "admin"](https://github.com/dimagi/commcare-hq/blob/022f199780d5db7da3adc2935c295014e7f21d85/corehq/apps/users/models.py#L459), looking at [one such instance](https://github.com/dimagi/commcare-hq/blob/022f199780d5db7da3adc2935c295014e7f21d85/corehq/apps/users/models.py#L734-L737) where this information is consumed. The logic to extract the role id is hence present in multiple places.
And we are doing a search here which does not have admin role, so simply using the ids works.
But I see that I have introduced a different behaviour here, not knowing that using role IDs was not done anywhere in HQ, even though using ids for search is a normal behaviour.
For consistency we can stick to using the qualified id always.

Reply:
Oh i see now that this search [originally was using id](https://github.com/dimagi/commcare-hq/pull/18038/files#diff-9d9075c8ac384b5859ac5aa7120651e5R1266) only. So not going to change this as of now but can take a follow up when I have more bandwidth to bring this in line with rest of HQ.

Reply:
Ah, that all makes sense, thank you for explaining.